src/FunctionRef.h

```diff
@@ -0,0 +1,62 @@
+
+#pragma once
+
+// Guard against macro new
+#pragma push_macro("new")
+#undef new
+
+// Declared only so it can be partially specialized
+template <class Signature>
+class cFunctionRef;
+
+/** A light-weight, type-erased reference to a function object.
+Non-owning, be mindful of lifetimes. */
+template <class Ret, class... Args>
+class cFunctionRef<Ret(Args...)>
```

Comment:
While a `std::function` would work, it is comparatively heavy-weight. It has to copy the object which might require heap allocations (depending on the implementations small functor optimisation limit).  This on the other hand has (or at least should have) no overhead compared to the inheritance method.

Not as much of a motivation but this will also support non-copyable functors.